code/raw_code/02_exploratory_analysis.R

```diff
@@ -1,212 +1,1077 @@
+                                              "No"))
+  
+  survey_df$Dangerous_Motorists <- factor(survey_df$Dangerous_Motorists,
+                                   levels = c("Yes",
+                                              "No"))
   
   ## ANALYSIS ###################################################################################################################
+  # Survey Statistics
+  total_surveys <- survey_df %>% 
+    summarize(Total_Survey_Count = n())
+  
+  goal_summary <- survey_df %>% distinct(Stop_ID) %>%
+    summarize(Total_Unique_Stops_surveyed = n(),
+              Percent_of_Goal = n()/ 2500)
+  
+  ## Top Overall and Weekly Surveyor Results
+  all_surveyors <- survey_df %>% distinct(Email_Masked, Stop_ID) %>%
+    group_by(Email_Masked) %>%
+    summarize(Total_Surveyed = n()) %>%
+    mutate(Email_Masked = ifelse(Email_Masked == "", "Not Specified", Email_Masked))
+  
+  ## Routes Survey Completion
+  route_stop_list <- route_data_2019 %>% distinct(route_short_name, route_long_name, stop_id) #%>%
+  
+  
+  distinct_stop_list <- survey_df %>% distinct(Stop_ID) %>%
+    mutate(Stop_ID_Join_Key = Stop_ID)
+  
+  route_stops_surveyed <- route_stop_list %>% left_join(distinct_stop_list, by = c("stop_id"="Stop_ID"), keep = TRUE) %>%
+    select(route_short_name,route_long_name, stop_id, Stop_ID_Join_Key) %>%
+    mutate(Stop_Survey_Count = ifelse(is.na(Stop_ID_Join_Key), 0, 1)) %>%
+    group_by(route_short_name, route_long_name) %>%
+    summarize(Total_Route_Stops = n(),
+              Total_Route_Stops_Surveyed = sum(Stop_Survey_Count),
+              Percent_Surveyed = Total_Route_Stops_Surveyed/ Total_Route_Stops)
+  
+  essential_service <- c("2","4", "5", "6", "12", "15", "19", "21", "26", "39", "40",
+                         "42", "49", "50", "51", "55", "60","66", "71", "73","74","78","81",
+                         "82", "83", "84", "86","87", "89", "95", "102", "107", "110",
+                         "111","115", "116", "117", "120", "121", "124", "125", "172", "178","180",
+                         "181","185", "186", "191", "192", "193", "196", "188", "816")
+  
+  total_stops_essential <- route_stop_list %>% 
+    dplyr::filter(route_short_name %in% essential_service) %>%
+    left_join(distinct_stop_list, by = c("stop_id"="Stop_ID"), keep = TRUE) %>%
+    distinct(stop_id, Stop_ID_Join_Key) %>%
+    mutate(Stop_Survey_Count = ifelse(is.na(Stop_ID_Join_Key), 0, 1)) %>%
+    summarize(Total_Stops = n(),
+              Total_Stops_Surveyed = sum(Stop_Survey_Count),
+              Percent_Surveyed = Total_Stops_Surveyed/ Total_Stops)
+  
+  
+  # Calculating County Percentages
+  county_results <- final_stop_list %>% left_join(distinct_stop_list, by = c("stop_id"="Stop_ID"), keep = TRUE) %>%
+    select(stop_id, Stop_ID_Join_Key, Stop_Lat, Stop_Lon) %>%
+    mutate(County = map.where(database="county", 
+                              Stop_Lon, Stop_Lat),
+           County = case_when(
```

Comment:
Potential to extract this mapping to a function (it seems to be shared with the clean data script).